iModelCore/ECDb/ECDb/ConcurrentQueryManagerImpl.cpp

```diff
@@ -80,9 +80,13 @@ std::shared_ptr<CachedQueryAdaptor> QueryAdaptorCache::TryGet(Utf8CP ecsql, bool
     ErrorListenerScope err_scope(const_cast<ECDb&>(m_conn.GetPrimaryDb()));
     if (usePrimaryConn)
         status = newCachedAdaptor->GetStatement().Prepare(m_conn.GetPrimaryDb(), ecsql, !suppressLogError);
-    else
-        status = newCachedAdaptor->GetStatement().Prepare(m_conn.GetPrimaryDb().Schemas(), m_conn.GetDb(), ecsql, !suppressLogError);
-
+    else {
+        if (m_doNotUsePrimaryConnToPrepare) {
+            status = newCachedAdaptor->GetStatement().Prepare(m_conn.GetDb(), ecsql, !suppressLogError);

```

Comment:
Should this use the overload taking a schemaManager? Its documentation says it's for multi-threading scenarios. If not, can you explain?

Reply:
No the whole point is not to use primary connection and see if we can prepare more in less by not competing for primary statement.